src/Autofac/ResolveRequest.cs

```diff
@@ -11,6 +11,11 @@ namespace Autofac;
 /// </summary>
 public class ResolveRequest
 {
+    /// <summary>
+    /// Shared constant value defining an empty set of parameters.
+    /// </summary>
+    internal static readonly IEnumerable<Parameter> NoParameters = Enumerable.Empty<Parameter>();
```

Comment:
I think as soon as it became public/internal, I didn't really want to introduce direct coupling between the `BaseGenericDelegateInvoker` and the `RegistrationExtensions` class. It felt like it introduced a dependency loop between the two classes in a way I didn't love.

I was close to creating a new `Parameters` internal class just to hold the constant but thought that may be overkill.

So, sort of because it was "kinda weird", I guess.